Automattic/vip-go-mu-plugins
# Add JP debug headers for broken connections

## Description

This conditionally adds JP debug headers for broken connections when the request comes from JP.

Before sending the debug headers, the request is verified as having come from JP to prevent this data from being returned to unauthorized sources.

Fixes #1300

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

This is tricky to fully test, because it relies on both failed JP requests, and the request coming from a valid JP IP. This gets pretty close:

1. Check out PR.
1. Add this code to `functions.php` to simulate a JP signature failure:

```
$error = new WP_Error( 'testing', 'test', array( 'signature_details' => 'hi' ) );

do_action( 'jetpack_verify_signature_error', $error );
```

1. Add your IP address to the list in `vip_is_jetpack_request()` in `vip-helpers/vip-utils.php`
1. Make a request to the site
1. Should see `x-jetpack-*` headers with the values from the fake error object
